src/main/java/jenkins/plugins/git/BuiltRevisionMapActionFactory.java

```diff
@@ -0,0 +1,29 @@
+package jenkins.plugins.git;
+
+import hudson.Extension;
+import hudson.model.AbstractProject;
+import hudson.model.Action;
+import hudson.model.TransientProjectActionFactory;
+import hudson.plugins.git.GitSCM;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+@Extension
+public class BuiltRevisionMapActionFactory extends TransientProjectActionFactory {
+
+    @Override
+    public Collection<? extends Action> createFor(AbstractProject target) {
+        if (target.getScm() instanceof GitSCM)
```

Comment:
No
please remember this is WiP
also, I hate multiple-scm. This plugin do impact all SCM plugin and introduce dependency on "_a proof-of-concept than a robust and fully functional component_". Such a plugin should not exist and would better have been designed as some API enhancement in jenkins-core.